db/silkworm/etl/collector.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+   Copyright 2020 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#include "collector.hpp"
+
+namespace silkworm::etl{
+void Collector::flush_buffer() {
+    buffer_.sort();
+    data_providers_.push_back(FileProvider(data_providers_.size()));
+    data_providers_.at(data_providers_.size()-1).write_buffer_to_disk(buffer_.get_entries());
```

Comment:
When inserting new class elements into a vector use `emplace_back` instead of creating new instance of the class which gets then copied into the vector.
Also ... in the second line you clearly want to access the last inserted element of the vector. Use `back()` to get the reference to it.

```suggestion
    data_providers_.emplace_back(data_providers_.size());
    data_providers_.back().write_buffer_to_disk(buffer_.get_entries());
```